src/tests/build.proj

```diff
@@ -274,6 +274,9 @@
       <BuildDir>$(IntermediateOutputPath)\iOSApps\$(Category)</BuildDir>
       <FinalPath>$(XUnitTestBinBase)$(CategoryWithSlash)\$(Category).app</FinalPath>
       <RuntimeComponents>diagnostics_tracing;marshal-ilgen</RuntimeComponents>
+      <DiagnosticPorts>127.0.0.1:9000,nosuspend,listen</DiagnosticPorts>
+      <RunAOTCompilation>True</RunAOTCompilation>
```

Comment:
I'm a little confused here; what happens if we are running iOSSimulator and both `RunAOTCompilation` and `MonoForceInterpreter` are true? Does this change how iOSSimulator tests are run?

Reply:
For iOSlike devices and arm based simulators, AOT is needed for the tests to build. (At least that was the case for library tests, which motivated this PR https://github.com/dotnet/runtime/pull/76551) When setting RunAOTCompilation and MonoForceInterpreter to true, what will happen is AOT will be ran, and if anything goes wrong, interp will be the fallback instead of JIT (which is important because JIT does not work on iOSlike platforms).
I believe the RunAOTCompilation is necessary here for arm architectures, but we can exclude x64 if thats preferred.